### PR TITLE
Fix Steam Input handling

### PR DESCRIPTION
**Pull request by offalynne**
**Update: See https://github.com/JujuAdams/Input/pull/267#issuecomment-1090676226**

Steam is now spoofing an Xbox controller on MacOS (as on Windows) instead of the former cleaner, dedicated, "Steam Virtual Controller" device (as on Linux). This appears to be a hasty/broken change because the virtual devices frequently show up in duplicate, and Steam itself seems to fail at recognizing them as virtualized, incorrectly enumerating them as additional mappable devices.

Fortunately (?) as of Big Sur, Apple has killed connectivity support for the real device being spoofed (Xbox 360 wired controller), so we can discard it outright via blacklist.

In addition, we can use envars including SDL hint `SDL_GAMECONTROLLER_IGNORE_DEVICES` to improve support for Steam Input's virtual controllers on Linux.

### CODE REVIEW

**Comment by offalynne** [comment]
On Mac, Steam virtual controllers are broken, so they're blocked.

On Linux, we block the virtual controller so when the game runs outside of Steam, the Steam Input desktop config will not interfere with normal gamepad operation. When run from Steam, native controllers are hidden, so we unblock the virtual controller (thx @nkrapivin).

**Comment by offalynne** [comment]
~~Good to go, thx again @nkrapivin~~

Still seeing device dupes in Steam according to @tabularelf.. back in with the type blocklist I guess.

**Comment by offalynne** [comment]
Annnd we're back again

**Comment by offalynne** [comment]
👁️  
image

brb

**Comment by offalynne** [comment]
Just as a summary note of these changes since the log is kind of hairy:

On Mac, no virtual controllers, access the original devices.

On Linux, no virtual controllers, *unless* Steam is running, in which case if necessary (overlay hook is misbehaving) block original devices Steam is using.

**Comment by offalynne** [comment]
Just as a summary note of these changes since the log is kind of hairy:

- Inlines Steam Input handling
- Corrects blocklist platform field typo
- On Mac, no virtual controllers, access original devices
- On Linux, no virtual controllers unless run thru Steam, in which case block designated device types if overlay is unloaded

File: scripts/__input_initialize/__input_initialize.gml [hunk 1]
```diff
@@ -69,6 +69,9 @@ function __input_initialize()
     //Struct to store all the keyboard keys we want to ignore
     global.__input_ignore_key_dict = {};
     
+    //Struct to store ignored gamepad types
+    global.__input_ignore_gamepad_types = {}
+
```
Comment (JujuAdams) [inline scripts/__input_initialize/__input_initialize.gml:74]: Is `global.__input_ignore_gamepad_types` used anywhere else? Can't spot it

Comment (JujuAdams) [reply scripts/__input_initialize/__input_initialize.gml:74]: ...apparently I am blind, carry on

**Comment by JujuAdams** [comment]
Thank you very much! Took me a few tries to get my head around Steam on Mac / Linux

**Pull request merged**